crates/supervisor/core/src/supervisor.rs

```diff
@@ -272,7 +273,7 @@ impl Supervisor {
     }
 
     fn init_l1_watcher(&self) -> Result<(), SupervisorError> {
-        let l1_rpc = RpcClient::new_http(self.config.l1_rpc.parse().unwrap());
+        let l1_rpc = Arc::new(RpcClient::new_http(self.config.l1_rpc.parse().unwrap()));
```

Comment:
`RpcClient` is lightweight, cloneable structure. It internally uses an `Arc<RpcClientInner>`, you do not need to wrap it in another `Arc`.